core/src/main/scala/org/bitcoins/core/protocol/script/ScriptPubKey.scala

```diff
@@ -696,9 +777,9 @@ object WitnessScriptPubKey {
 
     //we can also have a LockTimeScriptPubKey with a nested 0 public key multisig script, need to check that as well
     val bytes = BitcoinSUtil.toByteVector(asm)
-    val isMultiSig =
+    lazy val isMultiSig =
       MultiSignatureScriptPubKey.isMultiSignatureScriptPubKey(asm)
-    val isLockTimeSPK = {
+    lazy val isLockTimeSPK = {
```

Comment:
For the record, this was a bug where we were checking asm under the assumption that it was non-empty in cases when it was empty